test/finders/songs_finder_test.rb

```diff
@@ -34,6 +35,11 @@ class SongsFinderTest < ActionController::TestCase
   end
 
   test 'retrieve all returns many songs' do
-    assert(SongsFinder.retrieve_all.count > 7) 
+    assert(SongsFinder.retrieve_all.count > 7)
+  end
+
+  test 'should partially match' do
+    assert_includes(partial_word_results, songs(:hands_to_the_heaven))
+    assert_includes(partial_word_results, songs(:glorious_day))
   end
```

Comment:
How about one more test to check that the partial matching only works on the last term?